### PR TITLE
Greenbids Bidder Adapter

### PR DESCRIPTION
**Pull request by jeremy-greenbids**
## Type of change
 New bidder adapter 

**Module Name**: Greenbids Bidder Adapter  
**Module Type**: Bidder Adapter  
**Maintainer**:  tech@greenbids.ai

# Description

Use `greenbids` as bidder.

## AdUnits configuration example
```
    var adUnits = [{
      code: 'your-slot_1-div', //use exactly the same code as your slot div id.
      sizes: [[300, 250]],
      bids: [{
          bidder: 'greenbids',
          params: { 
              gbPlacementId: 12345,
          }
      }]
    },{
      code: 'your-slot_2-div', //use exactly the same code as your slot div id.
      sizes: [[600, 800]],
      bids: [{
          bidder: 'greenbids',
          params: { 
              placementId: 12345,
          }
      }]
    }];
```

### CODE REVIEW

File: modules/greenbidsBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+/**
+ * Retrieves the content of the page description meta tag.
+ *
+ * This function attempts to get the description from the top-level window's document.
+ * If it fails (e.g., due to cross-origin restrictions), it falls back to the current document.
+ * It looks for meta tags with either the name "description" or the property "og:description".
+ *
+ * @returns {string} The content of the description meta tag, or an empty string if not found.
+ */
+function getPageDescription() {
```
Comment (patmmccann) [inline modules/greenbidsBidAdapter.js:193]: There's a bunch of utility functions here that can be imported, could you move them all? 

Thanks!

**Comment by jeremy-greenbids** [comment]
@patmmccann the debate for dupplication as been settled on slack. Is it good for the rest please?

File: modules/greenbidsBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+      device: bidderRequest?.ortb2?.device || {},
+      deviceWidth: screen.width,
+      deviceHeight: screen.height,
+      devicePixelRatio: topWindow.devicePixelRatio,
+      screenOrientation: screen.orientation?.type,
+      historyLength: getHLen(),
+      viewportHeight: topWindow.visualViewport?.height,
+      viewportWidth: topWindow.visualViewport?.width,
+      hardwareConcurrency: getHC(),
+      deviceMemory: getDM(),
```
Comment (patmmccann) [inline modules/greenbidsBidAdapter.js:75]: we require that you add a disclosure to your documentation to use these api's

Comment (patmmccann) [reply modules/greenbidsBidAdapter.js:75]: see https://github.com/prebid/prebid.github.io/pull/5572

Comment (patmmccann) [reply modules/greenbidsBidAdapter.js:75]: and #12060

Comment (jeremy-greenbids) [reply modules/greenbidsBidAdapter.js]: I removed them, did not realised they would be for finger printing

**Comment by jeremy-greenbids** [comment]
I think it's good

**Comment by jeremy-greenbids** [comment]
Can you tell if it is good now please? :)

**Review by patmmccann — APPROVED** [review]

**Comment by patmmccann** [comment]
Welcome to the project!

**Pull request merged**